CondCore/RunInfoPlugins/plugins/RunInfo_PayloadInspector.cc

```diff
@@ -237,12 +236,12 @@ namespace {
         case RunInfoPI::m_min_current:
           return std::make_pair(isRealRun, payload.m_min_current);
         case RunInfoPI::m_BField:
-          return std::make_pair(isRealRun, fieldIntensity);
+          return std::make_pair(isRealRun, RunInfoPI::theBField(payload.m_avg_current));
         default:
           edm::LogWarning("LogicError") << "Unknown parameter: " << param;
           break;
       }
-
+      return std::make_pair(isRealRun, -1.0);
```

Comment:
Just to understand how this alleged optimization works, this can't be `return`ed in the `default` clause?

Reply:
it can be return in `default` clause but then (I think) you still have to add a return statement outside the `switch` block. 
I am also not sure what should be return in case of no-match/default clause.